lib/entity.js

```diff
@@ -21,27 +21,40 @@ module.exports = Entity;
  */
 
 function Entity(options){
-  this.protocol = window.location.protocol;
   this.options(options);
+  this.initialize();
 }
 
 /**
- * Get the storage.
- *
- * When .protocol is `file:` or `chrome-extension:`
- * the method will return the localstorage (store)
- * otherwise it will return the cookie.
+ * Initialize picks the storage.
  *
- * @return {Object}
+ * Checks to see if cookies can be set
+ * otherwise fallsback to localStorage.
+ */
+
+Entity.prototype.initialize = function(){
+  cookie.set('ajs:cookies', true);
+
+  // cookies are enabled.
+  if (cookie.get('ajs:cookies')) {
+    cookie.remove('ajs:cookies');
+    this._storage = cookie;
+    return;
+  }
+
+  // fallback to localstorage.
+  this._storage = store;
```

Comment:
Can local storage ever be unavailable? And should we fallback to just generating a new `anonymousId` every time in that case with another `temp` store (or similar)?